[Tan Jun Yi] iP

Tanjy55 opened the pull request.

Tanjy55 committed:
Renamed chatbot from "Duke" to "Tank"

Implemented an initial skeletal version of it that simply greets the user and exits.

Tanjy55 committed:
Added Echo function: When user inputs keyboard string, Tank echos string back.

Program terminates when user inputs "bye"

Tanjy55 committed:
Added the ability to store whatever text entered by the user and display them back to the user using the command "list".

More than 100 texts are able to be stored.

Tanjy55 committed:
Renamed list to tasks. Functionality of "list" command is preserved.

Implemented task tracking using Task class, with ability to mark and unmark tasks as done/not done.

Example input: "mark 1"
Example input: "unmark 1"

Modified text prompts to sound more friendly.

kevinlokewy commented on src/main/java/Tank.java:
```diff
@@ -0,0 +1,77 @@
+        String[] parts = line.split(" ");
+        int ArrayIndex = Integer.parseInt(parts[1]) - 1;
+        list.get(ArrayIndex).setDone();
+        System.out.println("____________________________________________________________");
+        System.out.println("\t Good job! I've marked this task done:");
+        System.out.println("\t\t" + list.get(ArrayIndex).getStatusIcon() + " " + list.get(ArrayIndex).description);
+        System.out.println("____________________________________________________________");
```
Do you think it's possible to reduce all these print lines into a method?

kevinlokewy commented on src/main/java/Tank.java:
```diff
@@ -0,0 +1,77 @@
+
+public class Tank {
+    static void displayList(ArrayList<Task> list) {
+        int counter = 1;
```
I think it makes it easier to read the code if its "taskCounter" instead of just "counter", so the reader is reminded what we're counting

kevinlokewy commented on src/main/java/Tank.java:
```diff
@@ -0,0 +1,77 @@
+
+    static void markTaskNotDone(ArrayList<Task> list, String line) {
+        String[] parts = line.split(" ");
+        int ArrayIndex = Integer.parseInt(parts[1]) - 1;
```
Retrieving the array index like this is kind of confusing at first glance, if we can use a method it might be more readable like getArrayIndex()

kevinlokewy commented on src/main/java/Tank.java:
```diff
@@ -0,0 +1,77 @@
+        System.out.println("What shall we talk about today?\n");
+        System.out.println("____________________________________________________________\n");
+
+        do {
+            line = in.nextLine();
+            if (line.equals(("list"))) {
+                displayList(listOfTasks);
+                continue;
+            }
+
+            if (line.contains(("unmark"))) {
+                markTaskNotDone(listOfTasks, line);
+                continue;
+            }
+
+            if (line.contains(("mark"))) {
+                markTaskDone(listOfTasks, line);
+                continue;
+            }
+
+            if (!line.equals("bye")) {
```
I think its better to do 'if-else' instead of 'ifs' because the programme will check through all the 'ifs', but for 'if-else' the programme will exit the loop once it meets one of the conditions.

kevinlokewy replied:
I think its better to do 'if-else' instead of 'ifs' because the programme will check through all the 'ifs', but for 'if-else' the programme will exit the loop once it meets one of the conditions.

mazh25 commented on src/main/java/Tank.java:
```diff
@@ -0,0 +1,77 @@
+                displayList(listOfTasks);
+                continue;
+            }
+
```
I have same idea. Maybe it is a better idea to use if-else rather than "if" only.

mazh25 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+        isDone = false;
+    }
+
+    public String getStatusIcon() {
```
I like this coding style. Using get- and set- to name a method makes code readable.

mazh25 commented on src/main/java/Tank.java:
```diff
@@ -0,0 +1,77 @@
+            System.out.println("\t" + counter + "." + item.getStatusIcon() + " " + item.description);
+            counter++;
+        }
+        System.out.println("____________________________________________________________");
```
Maybe you can find a better way to tackle the print lines.

Joannaj00 commented on src/main/java/Tank.java:
```diff
@@ -0,0 +1,77 @@
+
+    static void markTaskDone(ArrayList<Task> list, String line) {
+        String[] parts = line.split(" ");
+        int ArrayIndex = Integer.parseInt(parts[1]) - 1;
```
Variable name should be camel case. (ArrayIndex -> arrayIndex)

mazh25 reviewed (commented):
Overall, the coding standards are excellent. I had to look quite carefully to find even minor issues, which speaks to the high quality and attention to detail throughout the code.

Joannaj00 commented on src/main/java/Tank.java:
```diff
@@ -0,0 +1,77 @@
+        ArrayList<Task> listOfTasks = new ArrayList<>();
+        int numberOfTasks = 0;
+
+        System.out.println("____________________________________________________________");
```
"____________________________________________________________" could be a constant instead of printing every time!!

Joannaj00 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+        this.isDone = false;
+    }
+
+    public void setDone() {
```
good job making the method names verbs and camelCase!!

Joannaj00 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
good job making the boolean name sound like a boolean!!

Tanjy55 committed:
Majority changes were done to improve coding standards.
- turned dotted lines into constant to remove clutter
- extracted certain functions according to SLAP method
- fixed variable naming that were not in camelcase
- modified main method to switch case instead of multiple if clauses

Tanjy55 committed:
Applied regression fix to markTaskDone and markTaskNotDone
Added checkIndexValidity method to print error message and reject invalid user input
Modified greet message

Tanjy55 committed:
Expanded task feature set by implmenting subclass of Task (Todo, Deadlines, Events).

Corresponding constructors, modifications to main method, displaylist method have been updated.

Added help message such that user inputs that do not specify todo, deadline or event do not invoke Task class to be constructed directly by user.

Tanjy55 committed:
Use the input/output redirection technique to semi-automate the testing of Tank.

Tanjy55 committed:
Improved code quality.

Tanjy55 committed:
Added exception handling functions that do not terminate Tank instance.

Invalid commands, invalid inputs and invalid formats will be detected, and Tank will prompt user with appropriate fixes.

Tanjy55 committed:
Added delete Task function. User can now specify for example "delete 2" to delete item 2 in the list.

Tanjy55 committed:
Resolved bug whereby mark, unmark and delete will cause Tank to fail if number specified is exactly 1 more than the current number of tasks.

Tanjy55 committed:
Updated HELP_SHEET to include delete function

Tanjy55 committed:
Merge branch 'branch-Level-5'

Tanjy55 committed:
Added TankException class to use exceptions to handle errors. Corresponding methods have been updated to use try-catch to process exceptions thrown by TankException.

LuoYu-uwu commented on src/main/java/Tank.java:
```diff
@@ -0,0 +1,362 @@
+            return true;
+        }
+
+        String[] message = processDeadlineInput(line);
```
Perhaps consider using plural noun instead of message if it always contains multiple parts

LuoYu-uwu commented on src/main/java/Tank.java:
```diff
@@ -0,0 +1,362 @@
+        }
+    }
+
+    static boolean checkValidDeadline(String line) {
```
Perhaps can rename the function to sound more like booleans instead of verb

LuoYu-uwu reviewed (commented):
Overall, clean and well structured codes that follows SLAP well. good job!

Tanjy55 committed:
Put all classes into one package: tank

Minor edit to help message to resolve formatting issue

Tanjy55 committed:
Modified existing delete according to Level-6

Tanjy55 committed:
Added function to save file according to Level-7

File will be saved in project folder/data. Tank will autosave the file when user creates a task, marks or deletes it

Tanjy55 committed:
Updated readme

Tanjy55 committed:
Updated text ui test files

Tanjy55 committed:
Updated README from old duke template to tank

Tanjy55 committed:
Merge branch 'branch-Level-6'

Tanjy55 committed:
Merge branch 'branch-Level-7'. Resolved merge conflicts

# Conflicts:
#	src/main/java/tank/Tank.java

Tanjy55 committed:
Added and built JAR file

Tanjy55 committed:
Updated README due to incorrect display on github

Tanjy55 committed:
Update README.md

Tanjy55 committed:
Update README.md

Tanjy55 committed:
Made the code more OOP by overhauling entire code design

TextUi: deals with user interaction
TankStoreFile: deals with loading and saving tasks from storage.txt
Parser: deals with user command and formatting input string
TaskList: contains task list, task subclasses, setter and getter methods
Command: contains subclasses that override the execute command for various purposes such as add, delete, mark

Tanjy55 committed:
Modified command class and subclasses to NOT store taskList data.

Fixed bugs with filestorage corruption, and load method and not recording isDone status,

Tanjy55 committed:
Updated to fix file storage bug when using Ip.jar

Tanjy55 committed:
Added find command. Added updates to add and delete commands to fix minor UI regressions after A-MoreOOP

Tanjy55 committed:
Added functionality, deadline (by) and event (from, to) are LocalDateTime objects

Users now record the date and the specified format DD/MM/YYYY HHMM

Tanjy55 committed:
Updated UI elements internally

Tanjy55 committed:
Added JavaDoc comments to the code and minor change to messages.

Tanjy55 committed:
Updated docs readme.md, minor changes to ui elements.

Tanjy55 committed:
Updated text in docs userguide.

Tanjy55 committed:
Updated user guide.

Tanjy55 committed:
Updated automated testing files and updated minor changes to JavaDoc.

Tanjy55 committed:
Minor edits to find function to fix file history.

Tanjy55 committed:
Merge pull request #1 from Tanjy55/branch-A-JavaDoc

Updated automated testing files and updated minor changes to JavaDoc.

Tanjy55 committed:
Merge pull request #2 from Tanjy55/branch-Level-9

Minor edits to find function to fix file history.

Tanjy55 committed:
Fix download link for latest release in readme.
